src/functions/utils/rxjs-dynamic-duration.spec.ts

```diff
@@ -57,17 +70,24 @@ describe(rxjsDynamicDuration.name, () => {
 	it(
 		'adds 500ms on every new event, starting with 1s, resets to 1s after the 2nd event',
 		unitTest(() => {
+			// Clock | Timer | What?
+			//     0 |  1000 | a starts a 1000ms timer
+			//  1000 |  1500 | a emits, b starts 1500ms timer
+			//  2500 |  1000 | b emits, c starts a 1000ms timer because it's index 2
+			//  3500 |  1500 | c emits, d starts a 1500ms timer
+			//  5000 |  2000 | d emits, e starts a 2000ms timer
+			//  7000 |     - | e emits, completes
 			const initialDuration = 1000;
-			const source = 'a b c d e';
-			const expected = '1000ms a 1499ms b 1999ms c 999ms d 1499ms e';
+			const source = 'a b c d e |';
+			const expected = '1000ms a 1499ms b 999ms c 1499ms d 1999ms (e|)';
```

Comment:
I changed the implementation a bit, so that we were deciding the delay for _this_ notification, rather than the next notification to be more like regular `debounce`